server/util-server.js

```diff
@@ -105,18 +98,20 @@ exports.ping = async (hostname) => {
  */
 exports.pingAsync = function (hostname, ipv6 = false) {
     return new Promise((resolve, reject) => {
-        const ping = new Ping(hostname, {
-            ipv6
-        });
-
-        ping.send(function (err, ms, stdout) {
-            if (err) {
-                reject(err);
-            } else if (ms === null) {
-                reject(new Error(stdout));
+        ping.promise.probe(hostname, {
+            v6: ipv6,
+            min_reply: 3
+        }).then((res) => {
+            // If ping failed, it will set field to unknown
+            if (res.host === "unknown") {
```

Comment:
Are there boolean true/false result in the res?

As because I think the string `unknown` could be a valid hostname.

Reply:
Yeah, I see what you mean. There isn't a Boolean for if the hostname is resolved, however if I change the condition to be `! res.active && res.hostname === "unknown"` it would mean that the monitor wouldn't fail when a hostname is set to "unknown" however it would mean that a failing monitor with a hostname of "unknown" would always show the error message `Name or service not known`. Not sure if this is acceptable or not. Given the very small number of cases where someone might use a hostname of "unknown", I think it should be fine so long as we add a small note in the documentation.

Reply:
My thoughts:

```
if (alive) {
    // Success
} else {
   // Fail
   if (host == "unknown") {
       // Reason: Name or service not known
   }
   
   if (ping == "unknown") {
       // Reason: Timeout
   }
}
```

Reply:
I haven't checked. Maybe in fail case, we don't need to actually check the `host` or `ping` values?

```
if (alive) {
    // Success
} else {
   // Fail
   // Reason: res.output
}
```